### PR TITLE
[Merged by Bors] - chore(CategoryTheory/Limits/Fubini): Fubini theorem for colimits

### PR DESCRIPTION
**Pull request by robin-carlier**
Dualize all statements in `CategoryTheory/Limits/Fubini.lean`.

---

I did not try to be smart about this and I kept statement and proofs as close as possible to the ones for limits. Main differences are the addition of a few type annotations. Tagged this as chore rather than feat as this consisted mostly copy-pasting.

[Open in Gitpod](https://gitpod.io/from-referrer/)

### CODE REVIEW

File: Mathlib/CategoryTheory/Limits/Fubini.lean [hunk 1]
```diff
@@ -342,6 +562,48 @@ theorem limitCurrySwapCompLimIsoLimitCurryCompLim_inv_π_π {j} {k} :
...
+    colimit (curry.obj (Prod.swap K J ⋙ G) ⋙ colim) ≅ colimit (curry.obj G ⋙ colim) :=
+  calc
+    colimit (curry.obj (Prod.swap K J ⋙ G) ⋙ colim) ≅ colimit (Prod.swap K J ⋙ G) :=
+      (colimitIsoColimitCurryCompColim _).symm
+    _ ≅ colimit G := (HasColimit.isoOfEquivalence (Prod.braiding K J) (Iso.refl _))
+    _ ≅ colimit (curry.obj G ⋙ colim) := colimitIsoColimitCurryCompColim _
+
+@[simp]
+theorem colimitCurrySwapCompColimIsoColimitCurryCompColim_ι_ι_hom {j} {k} :
+    colimit.ι _ j ≫  colimit.ι (curry.obj (Prod.swap K J ⋙ G)⋙ colim) k ≫
```
Comment (erdOne) [inline Mathlib/CategoryTheory/Limits/Fubini.lean:584]:
```suggestion
    colimit.ι _ j ≫  colimit.ι (curry.obj (Prod.swap K J ⋙ G) ⋙ colim) k ≫
```

File: Mathlib/CategoryTheory/Limits/Fubini.lean [hunk 2]
```diff
@@ -342,6 +562,48 @@ theorem limitCurrySwapCompLimIsoLimitCurryCompLim_inv_π_π {j} {k} :
...
+      (colimitCurrySwapCompColimIsoColimitCurryCompColim G).hom =
+        (colimit.ι _ k ≫ colimit.ι (curry.obj G ⋙ colim) j : _ ⟶ colimit (curry.obj G⋙ colim)) := by
+  dsimp [colimitCurrySwapCompColimIsoColimitCurryCompColim]
+  slice_lhs 1 3 => simp only []
+  simp
+
+@[simp]
+theorem colimitCurrySwapCompColimIsoColimitCurryCompColim_ι_ι_inv {j} {k} :
+    colimit.ι _ k ≫  colimit.ι (curry.obj G⋙ colim) j ≫
```
Comment (erdOne) [inline Mathlib/CategoryTheory/Limits/Fubini.lean:593]:
```suggestion
    colimit.ι _ k ≫  colimit.ι (curry.obj G ⋙ colim) j ≫
```

File: Mathlib/CategoryTheory/Limits/Fubini.lean [hunk 3]
```diff
@@ -342,6 +562,48 @@ theorem limitCurrySwapCompLimIsoLimitCurryCompLim_inv_π_π {j} {k} :
...
+      (colimitCurrySwapCompColimIsoColimitCurryCompColim G).inv =
+        (colimit.ι _ j ≫
+          colimit.ι (curry.obj _ ⋙ colim) k :
+            _ ⟶ colimit (curry.obj (Prod.swap K J ⋙ G)⋙ colim)) := by
```
Comment (erdOne) [inline Mathlib/CategoryTheory/Limits/Fubini.lean:597]:
```suggestion
            _ ⟶ colimit (curry.obj (Prod.swap K J ⋙ G) ⋙ colim)) := by
```

**Review by erdOne — COMMENTED** [review]
Otherwise LGTM

File: Mathlib/CategoryTheory/Limits/Fubini.lean [hunk 4]
```diff
@@ -342,6 +562,48 @@ theorem limitCurrySwapCompLimIsoLimitCurryCompLim_inv_π_π {j} {k} :
...
+    colimit.ι _ j ≫  colimit.ι (curry.obj (Prod.swap K J ⋙ G) ⋙ colim) k ≫
+      (colimitCurrySwapCompColimIsoColimitCurryCompColim G).hom =
+        (colimit.ι _ k ≫ colimit.ι (curry.obj G ⋙ colim) j : _ ⟶ colimit (curry.obj G⋙ colim)) := by
+  dsimp [colimitCurrySwapCompColimIsoColimitCurryCompColim]
+  slice_lhs 1 3 => simp only []
+  simp
+
+@[simp]
+theorem colimitCurrySwapCompColimIsoColimitCurryCompColim_ι_ι_inv {j} {k} :
+    colimit.ι _ k ≫  colimit.ι (curry.obj G ⋙ colim) j ≫
```
Comment (erdOne) [inline Mathlib/CategoryTheory/Limits/Fubini.lean:593]:
```suggestion
    colimit.ι _ k ≫ colimit.ι (curry.obj G ⋙ colim) j ≫
```

File: Mathlib/CategoryTheory/Limits/Fubini.lean [hunk 5]
```diff
@@ -342,6 +562,48 @@ theorem limitCurrySwapCompLimIsoLimitCurryCompLim_inv_π_π {j} {k} :
...
+    colimit (curry.obj (Prod.swap K J ⋙ G) ⋙ colim) ≅ colimit (curry.obj G ⋙ colim) :=
+  calc
+    colimit (curry.obj (Prod.swap K J ⋙ G) ⋙ colim) ≅ colimit (Prod.swap K J ⋙ G) :=
+      (colimitIsoColimitCurryCompColim _).symm
+    _ ≅ colimit G := (HasColimit.isoOfEquivalence (Prod.braiding K J) (Iso.refl _))
+    _ ≅ colimit (curry.obj G ⋙ colim) := colimitIsoColimitCurryCompColim _
+
+@[simp]
+theorem colimitCurrySwapCompColimIsoColimitCurryCompColim_ι_ι_hom {j} {k} :
+    colimit.ι _ j ≫  colimit.ι (curry.obj (Prod.swap K J ⋙ G) ⋙ colim) k ≫
```
Comment (erdOne) [inline Mathlib/CategoryTheory/Limits/Fubini.lean:584]:
```suggestion
    colimit.ι _ j ≫ colimit.ι (curry.obj (Prod.swap K J ⋙ G) ⋙ colim) k ≫
```

**Review by erdOne — COMMENTED** [review]
Oops missed some.

**Comment by erdOne** [comment]
maintainer merge

**Review by jcommelin — APPROVED** [review]
Thanks :tada:

bors merge

**Pull request closed without merging**